packages/ra-ui-materialui/src/detail/Edit.spec.tsx

```diff
@@ -106,6 +107,27 @@ describe('<Edit />', () => {
         });
     });
 
+    it("shoudln't display the Edit aside while loading with the emptyWhileLoading prop", async () => {
+        render(<EmptyWhileLoading />);
+
+        expect(screen.queryByText('Book Edition')).not.toBeNull();
+        expect(
+            screen.queryByText('War and Peace, by Leo Tolstoy (1869)')
+        ).toBeNull();
+        expect(screen.queryByText('Something went wrong')).toBeNull();
+
+        await waitFor(
+            () => {
+                expect(
+                    screen.queryByText('War and Peace, by Leo Tolstoy (1869)')
+                ).not.toBeNull();
+            },
+            { timeout: 2500 }
+        );
```

Comment:
This will slow down the unit tests. Instead of relying on a timeout, inject a data provider with a suspended Promise that you resolve manually.